kloppy/infra/serializers/tracking/epts/metadata.py

```diff
@@ -156,6 +157,18 @@ def load_metadata(
     root = objectify.fromstring(metadata_file.read())
     metadata = root.find("Metadata")
 
+    provider_path = objectify.ObjectPath("Metadata.GlobalConfig.ProviderName")
+    provider_name = provider_path.find(metadata)
+
+    if provider != None and provider != provider_name:
+        warnings.warn(
```

Comment:
Couple of review comments:
1. Pythonic way of checking if a variable is not None is just ‘if variable’
2. You compare a string with a Provider type, they will never be the same. A test to make sure a warning is thrown could confirm this
3. What happens when a provider name is set that we don’t know? I expect a warning
4. @bdagnino when I supply provider as an argument I expect that I overrule whatever is in the file. In current implementation I get a warning and the one in the file is leading. What do you think that should happen?

Reply:
I slipped up with 2. in the test aswell so yeah that totally makes sense.

Reply:
To compare them right do we need like a dictionary of accepted names in the XML? because we have a different name in the Provider class enum than it would be in the XML. And where is the right place to put that dictionary?

Reply:
For now this could be just be a 
```python
if AAA

elif BBB

else:
  raise Exception(f"Unknown provider {provider_name}. Please create a ticket at https://github.com/PySport/kloppy/issues")
```

Reply:
This should be fine! Are you also fine with creating a separate function?

Reply:
Absolutely!

Reply:
@koenvo about point 4, I agree the one given as input should prevail. Regarding the warning, probably ok to leave it? Could be a bit annoying if you really know what you are doing, but on the other hand I can't think of a good reason to want to do that (input a different provider than the one on the file).

Reply:
Agree and let's go with the warning. People might read it :-)